src/index.ts

```diff
@@ -420,19 +422,48 @@ class HashinalsWalletConnectSDK {
     };
   }
 
+  
+  public async generatePrivateAndPublicKey(): Promise<{
+    privateKey: string;
+    publicKey: string;
+  }> {
+    this.ensureInitialized();
+    const privateKey = await PrivateKey.generateED25519Async();
+    const publicKey = privateKey.publicKey;
+    return {
+      privateKey: privateKey.toString(),
+      publicKey: publicKey.toString()
+    };
+  }
+
+  public async updateTopic(topicId: string, memo: string, adminKey: string): Promise<string> {
+     this.ensureInitialized();
+    let transaction = new TopicUpdateTransaction()
+      .setTopicId(TopicId.fromString(topicId))
+      .setTopicMemo(memo || "")
+      .freezeWith(client);
+      
+    // Convert the adminKey string back to a PrivateKey object
+    const privateKey = PrivateKey.fromString(adminKey);
+    const signedTx = await transaction.sign(privateKey);
+    
+    const receipt = await this.executeTransaction(signedTx);
```

Comment:
Since you're already signing the transaction, the signer should be disabled in `executeTransaction`